opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ServiceContextProvider.java

```diff
@@ -31,16 +41,27 @@
 
     private final AuthenticationSessionRepository authSessions;
     private final ServiceSessionRepository serviceSessions;
+    private final FacadeEncryptionService facadeEncryptionService;
+    private final EncryptionProperties properties;
+    private final SecretKeyService secretKeyService;
 
     @Transactional
     @SneakyThrows
     public <T extends FacadeServiceableGetter> ServiceContext<T> provide(T request) {
-        if (null == request.getFacadeServiceable()) {
+        FacadeServiceableRequest facadeServiceable = request.getFacadeServiceable();
+        if (null == facadeServiceable) {
             throw new IllegalArgumentException("No serviceable body");
         }
 
+        UUID serviceSessionId = facadeServiceable.getServiceSessionId();
+        String sessionPassword = facadeServiceable.getSessionPassword();
+        KeyDTO keyDTO = getSessionSecretKey(serviceSessionId, sessionPassword);
```

Comment:
This method should be in `extractOrCreateServiceSession`

Reply:
ok